src/OmegaSearchJoin.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace DivineOmega\LaravelOmegaSearch;
+
+use Illuminate\Support\Str;
+
+class OmegaSearchJoin
+{
+    protected $joinedTable;
+    protected $localTable;
+    protected $conditions = [];
+    protected $joinType;
+
+    /**
+     * OmegaSearchJoin constructor.
+     * @param string $joinedTable
+     * @param string $localTable
+     * @param string $joinType
+     */
+    public function __construct($localTable, $joinedTable, $joinType = 'JOIN')
+    {
+        $this->joinedTable = $joinedTable;
+        $this->localTable = $localTable;
+        $this->joinType = $joinType;
+    }
+
+    /**
+     * @param string $localTable
+     * @param string $joinedTable
+     * @param string $foreignKey
+     * @param string|null $localKey
+     * @return OmegaSearchJoin
+     */
+    public static function joinTableByForeignKey($localTable, $joinedTable, $foreignKey = 'id', $localKey = false)
+    {
+        if (!$localKey) {
+            $localKey = Str::singular($joinedTable).'_id';
+        }
+
+        $join = new self($localTable, $joinedTable, 'LEFT JOIN');
```

Comment:
Not necessarily a problem, but I'm curious why you're using `LEFT JOIN` here, but using `JOIN` in the object's constructor.

Reply:
In most cases this method will be used when a related model is being joined. The idea was to give the user the flexibility to query by the absence of data too too.